Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)

rr-codes opened the pull request:
#### b0f4981f866ed138e47f9dce34634727ac2d1314
<pre>
Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)
Need the bug URL (OOPS!).
rdar://101052130

Reviewed by NOBODY (OOPS!).

WIP.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::stopKeyboardScrolling):
(WebCore::EventHandler::beginKeyboardScrollGesture):
(WebCore::EventHandler::startKeyboardScrollAnimationOnDocument):
(WebCore::EventHandler::startKeyboardScrollAnimationOnRenderBoxLayer):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::requestStartKeyboardAnimation):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::requestStartKeyboardAnimation):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::boxSideForDirection): Deleted.
(WebCore::perpendicularAbsoluteUnitVector): Deleted.
(WebCore::KeyboardScrollingAnimator::updateKeyboardScrollPosition): Deleted.
(WebCore::KeyboardScrollingAnimator::beginKeyboardScrollGesture): Deleted.
(WebCore::farthestPointInDirection): Deleted.
(WebCore::KeyboardScrollingAnimator::stopKeyboardScrollAnimation): Deleted.
(WebCore::KeyboardScrollingAnimator::handleKeyUpEvent): Deleted.
(WebCore::KeyboardScrollingAnimator::stopScrollingImmediately): Deleted.
* Source/WebCore/platform/KeyboardScrollingAnimator.h:
* Source/WebCore/platform/KeyboardScrollingState.cpp: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/KeyboardScrollingState.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
(WebCore::KeyboardScrollingState::KeyboardScrollingState):
* Source/WebCore/platform/ScrollAnimation.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/ScrollAnimation.h:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp: Added.
(WebCore::ScrollAnimationKeyboard::ScrollAnimationKeyboard):
(WebCore::ScrollAnimationKeyboard::retargetActiveAnimation):
(WebCore::ScrollAnimationKeyboard::debugDescription const):
(WebCore::ScrollAnimationKeyboard::startKeyboardScroll):
(WebCore::scrollableDirectionsFromPosition):
(WebCore::boxSideForDirection):
(WebCore::perpendicularAbsoluteUnitVector):
(WebCore::ScrollAnimationKeyboard::animateScroll):
(WebCore::ScrollAnimationKeyboard::serviceAnimation):
* Source/WebCore/platform/ScrollAnimationKeyboard.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::ScrollAnimator):
(WebCore::ScrollAnimator::keyboardScrollToPosition):
* Source/WebCore/platform/ScrollAnimator.h:
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::beginKeyboardScroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::requestStartKeyboardAnimation):
* Source/WebCore/platform/ScrollingEffectsController.cpp:
(WebCore::ScrollingEffectsController::startKeyboardScroll):
(WebCore::ScrollingEffectsController::stopKeyboardScrolling):
(WebCore::ScrollingEffectsController::updateKeyboardScrollingAnimatingState):
* Source/WebCore/platform/ScrollingEffectsController.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::requestStartKeyboardAnimation):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
(WebKit::EventDispatcher::internalWheelEvent):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0f4981f866ed138e47f9dce34634727ac2d1314

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93185 "39 style errors") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2386 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23838 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102893 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/163180 "Hash b0f4981f for PR 5389 does not build (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97188 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2390 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30708 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85576 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99005 "Hash b0f4981f for PR 5389 does not build (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98854 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/76/builds/2390 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79655 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/36/builds/85576 "Hash b0f4981f for PR 5389 does not build (failure)") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/76/builds/2390 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/43/builds/23838 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/36/builds/85576 "Hash b0f4981f for PR 5389 does not build (failure)") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37099 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/43/builds/23838 "Hash b0f4981f for PR 5389 does not build (failure)") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34915 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/43/builds/23838 "Hash b0f4981f for PR 5389 does not build (failure)") | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38786 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/61/builds/79655 "Hash b0f4981f for PR 5389 does not build (failure)") | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40715 "Hash b0f4981f for PR 5389 does not build (failure)") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/43/builds/23838 "Hash b0f4981f for PR 5389 does not build (failure)") | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/67e98e7ae3e3ec92533ae9dda628beaf5d5c6b3a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/5389

rr-codes committed:
Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)
Need the bug URL (OOPS!).
rdar://101052130

Reviewed by NOBODY (OOPS!).

WIP.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::stopKeyboardScrolling):
(WebCore::EventHandler::beginKeyboardScrollGesture):
(WebCore::EventHandler::startKeyboardScrollAnimationOnDocument):
(WebCore::EventHandler::startKeyboardScrollAnimationOnRenderBoxLayer):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::requestStartKeyboardAnimation):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::requestStartKeyboardAnimation):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::boxSideForDirection): Deleted.
(WebCore::perpendicularAbsoluteUnitVector): Deleted.
(WebCore::KeyboardScrollingAnimator::updateKeyboardScrollPosition): Deleted.
(WebCore::KeyboardScrollingAnimator::beginKeyboardScrollGesture): Deleted.
(WebCore::farthestPointInDirection): Deleted.
(WebCore::KeyboardScrollingAnimator::stopKeyboardScrollAnimation): Deleted.
(WebCore::KeyboardScrollingAnimator::handleKeyUpEvent): Deleted.
(WebCore::KeyboardScrollingAnimator::stopScrollingImmediately): Deleted.
* Source/WebCore/platform/KeyboardScrollingAnimator.h:
* Source/WebCore/platform/KeyboardScrollingState.cpp: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/KeyboardScrollingState.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
(WebCore::KeyboardScrollingState::KeyboardScrollingState):
* Source/WebCore/platform/ScrollAnimation.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/ScrollAnimation.h:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp: Added.
(WebCore::ScrollAnimationKeyboard::ScrollAnimationKeyboard):
(WebCore::ScrollAnimationKeyboard::retargetActiveAnimation):
(WebCore::ScrollAnimationKeyboard::debugDescription const):
(WebCore::ScrollAnimationKeyboard::startKeyboardScroll):
(WebCore::scrollableDirectionsFromPosition):
(WebCore::boxSideForDirection):
(WebCore::perpendicularAbsoluteUnitVector):
(WebCore::ScrollAnimationKeyboard::animateScroll):
(WebCore::ScrollAnimationKeyboard::serviceAnimation):
* Source/WebCore/platform/ScrollAnimationKeyboard.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::ScrollAnimator):
(WebCore::ScrollAnimator::keyboardScrollToPosition):
* Source/WebCore/platform/ScrollAnimator.h:
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::beginKeyboardScroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::requestStartKeyboardAnimation):
* Source/WebCore/platform/ScrollingEffectsController.cpp:
(WebCore::ScrollingEffectsController::startKeyboardScroll):
(WebCore::ScrollingEffectsController::stopKeyboardScrolling):
(WebCore::ScrollingEffectsController::updateKeyboardScrollingAnimatingState):
* Source/WebCore/platform/ScrollingEffectsController.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::requestStartKeyboardAnimation):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
(WebKit::EventDispatcher::internalWheelEvent):

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b0f4981f866ed138e47f9dce34634727ac2d1314. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/5389
